lib/AST/Decl.cpp

```diff
@@ -2477,8 +2477,13 @@ getAccessScopeForFormalAccess(const ValueDecl *VD,
   const DeclContext *resultDC = VD->getDeclContext();
 
   while (!resultDC->isModuleScopeContext()) {
-    if (resultDC->isLocalContext() || access == AccessLevel::Private)
-      return AccessScope(resultDC, /*private*/true);
+    if (access == AccessLevel::Private)
+      return AccessScope(resultDC, /*private*/ true);
+
+    if (resultDC->isLocalContext()) {
+      return AccessScope(resultDC->getModuleScopeContext(),
+                         access == AccessLevel::Private);
```

Comment:
This is definitely wrong; true local variables (in function bodies) are not visible at the module scope. The only place to jump up a level is if resultDC is a TopLevelCodeDecl.

Reply:
Yes, in the test case I looked at it was a TopLevelCodeDecl. The assert that was firing here was in isAccessibleFrom() -- its the "does the fast path produce the same result as the access scope path" check. I'll change this to check for TopLevelCodeDecl instead of isLocalContext().